pulsar/src/main/java/io/micronaut/pulsar/processor/DefaultListener.java

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright 2017-2020 original authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.micronaut.pulsar.processor;
+
+import io.micronaut.context.annotation.Requires;
+import io.micronaut.inject.ExecutableMethod;
+import org.apache.pulsar.client.api.Consumer;
+import org.apache.pulsar.client.api.Message;
+
+/**
+ *
+ */
+@Requires(missingBeans = MessageListenerResolver.class)
+@SuppressWarnings({"rawtypes", "unchecked"})
+public class DefaultListener implements MessageListenerResolver {
+
+    private final boolean useMessageWrapper;
+    private final ExecutableMethod<Object, ?> method;
+    private final int consumerIndex;
+    private final Object invoker;
+
+    public DefaultListener(ExecutableMethod method, boolean useMessageWrapper, int consumerIndex, Object invoker) {
```

Comment:
Is that DefaultListener resp. MessageListenerResolver used at all?

Reply:
It's main bean that takes in data from Pulsar and calls functions declared as `@PulsarConsumer`. It provides flexibility to have different order in function parameters, completely ignore `org.apache.pulsar.client.api.Consumer` parameter, or even request message body to be extracted from default Pulsar `org.apache.pulsar.client.api.Message`. It also servers for developers to extend `MessageListenerResolver` to override this default behaviour in their own preffered way. We can force them to use default behaviour of MessageListener from Pulsar but I think most will prefer having option to ignore everything except from the actual body and having the option to implement something themselves.
Maybe MessageListenerResolver can be avoided and force user to only use our own implementation which is flexible enough and can be extended to support injecting more parameters like extra beans and such.